### PR TITLE
luaengine: add machine_flags library

### PR DESCRIPTION
**Pull request by vadosnaprimer**

### CODE REVIEW

**Comment by cracyc** [comment]
Why return a table here vs making each a separate property?

**Comment by vadosnaprimer** [comment]
I don't particularly need this as a separate table. Would you prefer just having them as game_driver fields?

**Comment by cracyc** [comment]
IMO it would be cleaner that way since the extra indirection isn't necessary.

**Comment by vadosnaprimer** [comment]
Done. I hope I spaced them correctly.

File: src/frontend/mame/luaengine.cpp [hunk 1]
```diff
@@ -1283,6 +1283,18 @@ void lua_engine::initialize()
  * driver.manufacturer
  * driver.compatible_with
  * driver.default_layout
+ * driver.orientation - screen rotation degree (rot0/90/180/270)
```
Comment (cuavas) [inline src/frontend/mame/luaengine.cpp]: This isn't necessarily going to give you a meaningful result.  Individual screens can also have rotation flags, and some systems have multiple screens with different orientations.

Comment (vadosnaprimer) [reply src/frontend/mame/luaengine.cpp]: You're right, it looks like a legacy flag obsoleted by `screen_device::orientation()`. Should I remove it? Seems to be easier to commit straight to master for whoever has access than setting up a new PR (unless my updates require such a clean-up PR).

**Pull request merged**